From dfe0e8702d5a66e9d32231dccff3bc17e5fd61d5 Mon Sep 17 00:00:00 2001
From: Xiaowei Bao <xiaowei.bao@nxp.com>
Date: Wed, 9 Jan 2019 18:14:22 +0800
Subject: [PATCH 020/235] Revert "misc: pci_endpoint_test: fix the call trace
 when the module removed."

This reverts commit 2e0ebdc2b8fd7148cb005b362d06075b968df937.
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/misc/pci_endpoint_test.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/misc/pci_endpoint_test.c b/drivers/misc/pci_endpoint_test.c
index faf0801c0297..8398d176b8d4 100644
--- a/drivers/misc/pci_endpoint_test.c
+++ b/drivers/misc/pci_endpoint_test.c
@@ -98,7 +98,6 @@ struct pci_endpoint_test {
 	struct miscdevice miscdev;
 	enum pci_barno test_reg_bar;
 	size_t alignment;
-	char name[20];
 };
 
 struct pci_endpoint_test_data {
@@ -467,6 +466,7 @@ static int pci_endpoint_test_probe(struct pci_dev *pdev,
 	int err;
 	int irq = 0;
 	int id;
+	char name[20];
 	enum pci_barno bar;
 	void __iomem *base;
 	struct device *dev = &pdev->dev;
@@ -561,10 +561,10 @@ static int pci_endpoint_test_probe(struct pci_dev *pdev,
 		goto err_iounmap;
 	}
 
-	snprintf(test->name, sizeof(test->name), DRV_MODULE_NAME ".%d", id);
+	snprintf(name, sizeof(name), DRV_MODULE_NAME ".%d", id);
 	misc_device = &test->miscdev;
 	misc_device->minor = MISC_DYNAMIC_MINOR;
-	misc_device->name = test->name;
+	misc_device->name = name;
 	if (!misc_device->name) {
 		err = -ENOMEM;
 		goto err_ida_remove;
-- 
2.17.1

